### PR TITLE
WIP - Set a services_prefix to prefix all service names with a Datadog::Tracer

### PR DESCRIPTION
**Pull request by bramswenson**
Looking for feedback and adding to the options for solving #142

### CODE REVIEW

**Comment by bentheax** [comment]
It would be good to make the solution work equally well for Sinatra, or for a previously unknown environment ((Hanami)[http://hanamirb.org]?).

**Comment by emaiax** [comment]
Looking forward to see this working.

File: lib/ddtrace/tracer.rb [hunk 1]
```diff
@@ -100,13 +101,19 @@ def configure(options = {})
     #
     #   tracer.set_service_info('web-application', 'rails', 'web')
     def set_service_info(service, app, app_type)
-      @services[service] = {
+      service_key = if services_prefix && service.start_with?(services_prefix)
```
Comment (ufoot) [inline lib/ddtrace/tracer.rb]: I think this is "not enough" as `set_service_info` is not called that often. In fact, you could trace a lot of code and never call it as it's only used to update the metadata of the service (eg: say "hey, this service is a database"). So I think to complete this patch one would need to hook somewhere in the `service=` method of spans (which is implemented with an `attr_accessor` right now). But as you mention, this is related to #142 so let's save this for now I think.

**Comment by delner** [comment]
Hey @bramswenson, thanks for the pull request!

Improving service names has been one of the focuses for the 0.11.0 release. Although the current plan is not to support service prefixes in such an explicit manner, we did want to address this by two other means.

1. Improving our default service names for a better out of the box experience.
2. An new configuration API that grants the ability to customize service names for many of the integrations currently supported by the library.

The second of these options might be useful for addressing your particular issue, and would let you define your own prefixes. For example:

```ruby
prefix = 'foo'

Datadog.configure do |c|
  c.use :rails, service_name: "#{prefix}-app"
  c.use :sidekiq, service_name: "#{prefix}-sidekiq"
end
```

This might solve the issue, so I'm going to close this pull request for now. Please feel free to check it out, and respond here if you'd like to explore this issue further!

Thanks again for your contribution!

**Pull request closed without merging**